app/uk/gov/hmrc/payetaxcalculatorfrontend/quickmodel/TaxResult.scala

```diff
@@ -79,6 +79,35 @@ object TaxResult {
       case _ => -1
     }
 
+  def incomeTax(breakdown : TaxBreakdown): BigDecimal = {
+    val maxTaxAmount = breakdown.maxTaxAmount
+    val stdIncomeTax: BigDecimal = extractIncomeTax(breakdown)
+
+    incomeTax(maxTaxAmount: BigDecimal, stdIncomeTax)
+  }
+
+  def incomeTax(maxTaxAmount: BigDecimal, stdIncomeTax: BigDecimal): BigDecimal = {
+    if(maxTaxAmount >= 0) stdIncomeTax min maxTaxAmount
+    else stdIncomeTax
+  }
+
+  def extractIncomeTax( breakdown : TaxBreakdown): BigDecimal = {
+    breakdown.taxCategories.find(_.taxType == "incomeTax").map(_.total)
+      .getOrElse(throw new AssertionError("Unknown Engine change"))
```

Comment:
I would not use an Error at all but an Either with an type explaining the issue. This would mean that the compiler forces we handle the error and do so with more explicit code.